packages/progress/src/ux-progress.ts

```diff
@@ -0,0 +1,97 @@
+   25.0001%   { stroke-dashoffset: START_VALUE;  transform: rotate(270deg); }
+   37.5%   { stroke-dashoffset: END_VALUE;    transform: rotate(270deg); }
+   37.5001%  { stroke-dashoffset: END_VALUE;    transform: rotateX(180deg) rotate(161.5deg); }
+   50%     { stroke-dashoffset: START_VALUE;  transform: rotateX(180deg) rotate(161.5deg); }
+   50.0001%  { stroke-dashoffset: START_VALUE;  transform: rotate(180deg); }
+   62.5%   { stroke-dashoffset: END_VALUE;    transform: rotate(180deg); }
+   62.5001%  { stroke-dashoffset: END_VALUE;    transform: rotateX(180deg) rotate(251.5deg); }
+   75%     { stroke-dashoffset: START_VALUE;  transform: rotateX(180deg) rotate(251.5deg); }
+   75.0001%  { stroke-dashoffset: START_VALUE;  transform: rotate(90deg); }
+   87.5%   { stroke-dashoffset: END_VALUE;    transform: rotate(90deg); }
+   87.5001%  { stroke-dashoffset: END_VALUE;    transform: rotateX(180deg) rotate(341.5deg); }
+   100%    { stroke-dashoffset: START_VALUE;  transform: rotateX(180deg) rotate(341.5deg); }
+ }`;
+
+@inject(Element, StyleEngine)
+@customElement('ux-progress')
+@useView(PLATFORM.moduleName('./ux-progress.html'))
+export class UxProgress {
+  constructor(public element: HTMLElement, private styleEngine: StyleEngine) { }
+
+  public strokeDashOffset: number;
+  public valueNumber: number | null | undefined;
+  public viewBox: string;
+  public animationTemplate: string;
+  public strokeCircumference: number;
+
+  @bindable
+  public value: number | string;
+  public valueChanged() {
+    this.valueNumber = normalizeNumberAttribute(this.value);
+    this.update();
+  }
+
+  @bindable
+  public theme: UxProgressTheme;
+  public themeChanged(newValue: UxProgressTheme) {
+    if (newValue !== null && !newValue.themeKey) {
+      newValue.themeKey = 'progress';
+    }
+
+    this.styleEngine.applyTheme(newValue, this.element);
+  }
+
+  private diameterNumber: number | null | undefined;
+  @bindable
+  public diameter: number | string = 100;
+  public diameterChanged() {
+    this.diameterNumber = normalizeNumberAttribute(this.diameter);
+    this.update();
+  }
+
+  private strokeWidthNumber: number | null | undefined;
+  @bindable
+  public strokeWidth: number | string = 10;
+  public strokeWidthChanged() {
+    this.strokeWidthNumber = normalizeNumberAttribute(this.strokeWidth);
+    this.update();
+  }
+
+  public bind() {
+    this.valueNumber = normalizeNumberAttribute(this.value);
+    this.diameterNumber = normalizeNumberAttribute(this.diameter);
+    this.strokeWidthNumber = normalizeNumberAttribute(this.strokeWidth);
+    this.update();
+  }
+
+  private update() {
+    this.strokeCircumference = ((this.diameterNumber ?? 0) - (this.strokeWidthNumber ?? 0)) * 3.14;
+    if (this.valueNumber === undefined || this.valueNumber === null) {
+      this.strokeDashOffset = 0;
+      const styleId = `ux-progress-animation-template-${this.diameter}-${this.strokeWidthNumber}`;

```

Comment:
does this mean the old styleId will be left orphaned in the document?

Reply:
Unless we track which progress instance uses which style. Is it worth it?
Is there an API which does it?

Reply:
Yes, we have inject style api for this. You can pass the id of the style element in 2nd argument so that it will reuse an existing one if already injected

Reply:
That API is broken - https://github.com/aurelia/ux/issues/260
Btw, that API does not hold element references either.

Reply:
That API doesn't need to hold element reference, it just need to use the same ID for the `<style/>` element every time it's called. Btw, can you please help try my suggestion in that issue? Adding a `text-loader!` in front of the module name, and load the module with only the path?